src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1040,11 +1037,7 @@ void SolverInterfaceImpl::mapReadDataTo(
       if (context.getMeshID() != toMeshID) {
         continue;
       }
-      context.resetToData();
-      PRECICE_DEBUG("Map data \"{}\" to mesh \"{}\"", context.getDataName(), context.getMeshName());
-      PRECICE_ASSERT(mappingContext.mapping == context.mappingContext().mapping);
```

Comment:
Why removing the Assert here?

Reply:
No specific reason. I added it again and I guess we can keep it. See fc50bcb81df7829d0aef598643de7de0cb6278fe. It might still be a good idea to remove it at a later point in time when more functionality is moved inside `DataContext`.